src/AI/CoreLogic.pm

```diff
@@ -2692,6 +2692,7 @@ sub processPartySkillUse {
 		for (my $i = 0; exists $config{"partySkill_$i"}; $i++) {
 			next if (!$config{"partySkill_$i"});
 			next unless checkSelfCondition("partySkill_$i");
+			$ai_v{"partySkill_$i"."_time"} = time;
```

Comment:
is this the right  place? what will happen  if succes through checkSelfCondition but fail on checkPlayerCondition ?